platform/linux-generic/odp_schedule_scalable.c

```diff
@@ -1007,6 +1007,8 @@ static void schedule_order_lock(unsigned lock_index)
 		       monitor32(&rctx->rwin->olock[lock_index],
 				 __ATOMIC_ACQUIRE) != rctx->sn)
 			doze();
+	rctx->rwin->lock_index = lock_index;
+
```

Comment:
This needs to go after the `if` statement, not inside it, since the whole statement is skipped if the lock is immediately available--it's only there to wait for the lock.